### PR TITLE
Add copyFileUntracked

### PR DESCRIPTION
**Pull request by izgzhen**

### CODE REVIEW

**Comment by izgzhen** [comment]
Should I fix the bit in `Libffi`? `copyFile` is somewhat unreasonable, but not really affecting the last issue.

**Comment by snowleopard** [comment]
Thanks!

Yes, let's fix the missing build result in the `libffi` rule and switch to `copyFileUntracked`.

**Comment by snowleopard** [comment]
I mean, we could probably do it within this PR.

File: src/Util.hs [hunk 1]
```diff
@@ -98,6 +98,14 @@ copyFile source target = do
     putProgressInfo $ renderAction "Copy file" source target
     copyFileChanged source target
 
+-- Same as copyFile, but not tracking the source as a build dependency
+copyFileUntracked :: FilePath -> FilePath -> Action ()
+copyFileUntracked source target = do
+    let dir = takeDirectory target
+    unlessM (liftIO $ IO.doesDirectoryExist dir) $ createDirectory dir
```
Comment (ndmitchell) [inline src/Util.hs]: Why not just createDirectoryIfMissing? That also works recursively, which is usually what you want.

Comment (snowleopard) [inline src/Util.hs]: Aha, looks like it fits indeed.

Are you suggesting to replace the line with `createDirectoryIfMissing True dir`?

Comment (ndmitchell) [inline src/Util.hs]: Yes. That's what Shake itself uses.

Comment (snowleopard) [inline src/Util.hs]: @izgzhen Let's use `createDirectoryIfMissing True dir` indeed. Please also amend the `copyFile` function.

**Comment by izgzhen** [comment]
I am ready on this, any more comments?

**Comment by snowleopard** [comment]
Many thanks @izgzhen! Merged now.

**Pull request merged**